File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int>mp;
        int j = 0;
        int n = s.size();
        int maxlen = 0;
        for(int i = 0; i < n; i++){
            while(mp.find(s[i]) != mp.end()) {
                mp[s[j]]--;
                if(mp[s[j]] == 0){
                    mp.erase(s[j]);
                }
                j++;
            }
            mp[s[i]]++;
            maxlen = max(maxlen,i - j +1);
        }
        return maxlen;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java

                   class Solution {
                        public int lengthOfLongestSubstring(String s) {
                                    int[] last = new int[256];
                                            java.util.Arrays.fill(last, -1);
                                                    int maxLen = 0;
                                                            int left = 0;
                                                                    for (int right = 0; right < s.length(); right++) {
                                                                                    char c = s.charAt(right);
                                                                                                if (last[c] >= left) {
                                                                                                                    left = last[c] + 1;
                                                                                                }
                                                                                                            last[c] = right;
                                                                                                                        maxLen = Math.max(maxLen, right - left + 1);
                                                                    }
                                                                            return maxLen;
                        }
                   }
                                                                                              
